sim/common/shared/shared_utils.go

```diff
@@ -97,14 +96,7 @@ func factory_StatBonusEffect(config ProcStatBonusEffect, extraSpell func(agent c
 
 	effectFn(effectID, func(agent core.Agent) {
 		character := agent.GetCharacter()
-		itemSwapIsEnabled := character.ItemSwap.IsEnabled()
-		itemExistsInMainEquip := true
-
-		var eligibleSlotsForItem []proto.ItemSlot
-		if !isEnchant {
-			eligibleSlotsForItem = character.ItemSwap.EligibleSlotsForItem(effectID)
-			itemExistsInMainEquip = !itemSwapIsEnabled || character.ItemSwap.ItemExistsInMainEquip(effectID, eligibleSlotsForItem)
-		}
+		eligibleSlots := core.Ternary(isEnchant, character.ItemSwap.EligibleSlotsForEffect(effectID), character.ItemSwap.EligibleSlotsForItem(effectID))
 
```

Comment:
This will call both functions needlessly, it's more efficient to do this instead:
```
var eligibleSlots []proto.ItemSlot

if isEnchant {
    eligibleSlots = character.ItemSwap.EligibleSlotsForEffect(effectID)
} else {
    eligibleSlots = character.ItemSwap.EligibleSlotsForItem(effectID)
}
```

Reply:
I wasn't sure if it does it lazy or not